File: src/main/java/com/bossymr/rapid/ide/execution/configurations/RapidRunConfigurationOptions.java
package com.bossymr.rapid.ide.execution.configurations;

import com.intellij.execution.configurations.ModuleBasedConfigurationOptions;
import com.intellij.openapi.components.StoredProperty;
import org.jetbrains.annotations.Nullable;

public class RapidRunConfigurationOptions extends ModuleBasedConfigurationOptions {

    /**
     * The name of the task on the robot to execute.
     */
    private final StoredProperty<String> taskName = string("").provideDelegate(this, "taskName");

    /**
     * The path to the robot.
     */
    private final StoredProperty<String> robotPath = string("").provideDelegate(this, "robotPath");

    public @Nullable String getTaskName() {
        return taskName.getValue(this);
    }

    public void setTaskName(@Nullable String taskName) {
        this.taskName.setValue(this, taskName);
    }

    public @Nullable String getRobotPath() {
        return robotPath.getValue(this);
    }

    public void setRobotPath(@Nullable String robotPath) {
        this.robotPath.setValue(this, robotPath);
    }
}

File: src/main/java/com/bossymr/rapid/ide/execution/configurations/RapidRunConfigurationOptions.kt
package com.bossymr.rapid.ide.execution.configurations

import com.intellij.execution.configurations.LocatableRunConfigurationOptions
import com.intellij.openapi.components.BaseState
import com.intellij.util.xmlb.annotations.Attribute
import com.intellij.util.xmlb.annotations.Property
import com.intellij.util.xmlb.annotations.XCollection

class TaskState() : BaseState() {

    constructor(name: String?, isEnabled: Boolean, moduleName: String?) : this() {
        this.name = name
        this.isEnabled = isEnabled
        this.moduleName = moduleName
    }

    @get:Attribute("name")
    var name by string()

    @get:Attribute("enabled")
    var isEnabled by property(true)

    @get:Attribute("module")
    var moduleName by string()
}

class RapidRunConfigurationOptions : LocatableRunConfigurationOptions() {

    @get:Property(surroundWithTag = false)
    @get:XCollection
    var robotTasks by list<TaskState>()

    @get:Attribute("path")
    var robotPath by string("")

}